### PR TITLE
Add threshold support to `BinaryAccuracy`

### PR DESCRIPTION
**Pull request by aaarrti**
Hi, 
I've recently found out that Keras V3 is missing support for `threshold` in `BinaryAccuracy`, 
in contrast to old `tf_keras` (https://github.com/keras-team/tf-keras/blob/master/tf_keras/metrics/accuracy_metrics.py#L113) 
The docstring was already present
```
threshold: (Optional) Float representing the threshold for deciding
        whether prediction values are 1 or 0.
```

### Changes
- [x] Add `threshold` argument
- [x] Add test case for `threshold argument`

P.S. I did not create an issue for this, since this is a tiny one 🤷

### CODE REVIEW

File: keras/metrics/accuracy_metrics.py [hunk 1]
```diff
@@ -113,11 +113,14 @@ class BinaryAccuracy(reduction_metrics.MeanMetricWrapper):
     ```
     """
 
-    def __init__(self, name="binary_accuracy", dtype=None):
-        super().__init__(fn=binary_accuracy, name=name, dtype=dtype)
+    def __init__(self, name="binary_accuracy", dtype=None, threshold=0.5):
+        if threshold is not None and (threshold <= 0 or threshold >= 1):
+            raise ValueError(f"`threshold` must be in interval (0, 1), but found: {threshold}")
```
Comment (fchollet) [inline keras/metrics/accuracy_metrics.py:118]: Standardize the message:

```
Invalid value for argument `threshold`. Expected a value in internal [0, 1]. Received: threshold={threshold}
```

File: keras/metrics/accuracy_metrics_test.py [hunk 2]
```diff
@@ -80,6 +80,34 @@ def test_weighted(self):
...
+        bin_acc_obj_1.update_state(y_true, y_pred)
+        bin_acc_obj_2.update_state(y_true, y_pred)
+        result_1 = bin_acc_obj_1.result()
+        result_2 = bin_acc_obj_2.result()
+        
+        # Higher threshold must result in lower measured accuracy.
+        np.testing.assert_array_less(result_2, result_1)
+    
+    def test_invalid_threshold(self):
+        self.assertRaises(
```
Comment (fchollet) [inline keras/metrics/accuracy_metrics_test.py:103]: Use `assertRaisesRegex`

File: keras/metrics/accuracy_metrics_test.py [hunk 3]
```diff
@@ -80,6 +80,34 @@ def test_weighted(self):
...
+        y_true = np.array([[1], [1], [0], [0]])
+        y_pred = np.array([[0.98], [0.5], [0.1], [0.2]])
+        
+        bin_acc_obj_1.update_state(y_true, y_pred)
+        bin_acc_obj_2.update_state(y_true, y_pred)
+        result_1 = bin_acc_obj_1.result()
+        result_2 = bin_acc_obj_2.result()
+        
+        # Higher threshold must result in lower measured accuracy.
+        np.testing.assert_array_less(result_2, result_1)
```
Comment (fchollet) [inline keras/metrics/accuracy_metrics_test.py:100]: Use an `assertAllClose` statement with hardcoded reference values.

**Review by fchollet — COMMENTED** [review]
Thank you for the PR!

To format the code, run `sh shell/format.sh`.

File: keras/metrics/accuracy_metrics.py (hunk 1 above)
Comment (aaarrti) [reply keras/metrics/accuracy_metrics.py]: The notation `[0, 1]` typically represent a closed interval (0 and 1 included).
0 and 1 are invalid values for `threshold`, so I'd rather use the open interval notation `(0, 1)`. I've changed the message to: 
```python
f"Invalid value for argument `threshold`. Expected a value in internal (0, 1). Received: threshold={threshold}"
```

File: keras/metrics/accuracy_metrics_test.py (hunk 2 above)
Comment (aaarrti) [reply keras/metrics/accuracy_metrics_test.py]: done

File: keras/metrics/accuracy_metrics_test.py (hunk 3 above)
Comment (aaarrti) [reply keras/metrics/accuracy_metrics_test.py]: done

**Comment by aaarrti** [comment]
> Thank you for the PR!
> 
> To format the code, run `sh shell/format.sh`.

done

**Review by fchollet — APPROVED** [review]
Thank you! Will fix remaining issues.

**Pull request merged**